client/src/components/Calendar/Views/WeeklyView.tsx

```diff
@@ -0,0 +1,90 @@
+import { generateWeekDates } from '../../../utils/scheduleUtils';
+import { format } from 'date-fns';
+import { WEEK_DAYS, CALENDAR_HOURS, TODAY } from '../../../utils/constants';
+import { cn } from '../../../utils/functions';
+import { isSameDay } from 'date-fns';
+import { useMemo } from 'react';
+
+type Props = {
+  initialDisplayDate: Date;
+};
+
+// There are 168 hours in a week.
+// We are displaying 2 slots per hour so that's 168 * 2 = 336
+// We have an additional row at the top so that's 336 + 7 = 343
+const NUMBER_OF_SLOTS = 343;
+
+const LIGHT_MODE_BORDER = 'border-gray-200';
+const DARK_MODE_BORDER = 'dark:border-gray-700';
+
```

Comment:
Figured I'd leave this constants here cause they're only being used by this component.